packages/blocks/src/api/raw-handling/paste-handler.js

```diff
@@ -127,9 +127,14 @@ export function pasteHandler( { HTML = '', plainText = '', mode = 'AUTO', tagNam
 	// First of all, strip any meta tags.
 	HTML = HTML.replace( /<meta[^>]+>/, '' );
 
-	// If we detect block delimiters, parse entirely as blocks.
-	if ( mode !== 'INLINE' && HTML.indexOf( '<!-- wp:' ) !== -1 ) {
-		return parseWithGrammar( HTML );
+	// If we detect block delimiters in HTML, parse entirely as blocks.
+	if ( mode !== 'INLINE' ) {
+		// Check plain text if there is no HTML.
+		const content = HTML ? HTML : plainText;
```

Comment:
I was hoping to find some information about when and why `clipboardData` would provide values for `text/plain` vs. `text/html`, and whether they're mutually exclusive, but the specification appears to be a bit ambiguous.

https://html.spec.whatwg.org/multipage/dnd.html#dom-datatransfer-getdata

I suppose if (a) the implementation favors HTML and (b) the non-existent form would be generated as an empty string (falsey), then this should be considered correct.

Reply:
I see it like this: there _should_ always be a plain text version of the pasted data, and if there's HTML, also an HTML version.

Because the browser might still provide an HTML version, even if there's no HTML, we check for it (as linked above): 
https://github.com/WordPress/gutenberg/pull/14536/files#diff-7703746fe73380c92b8bfe2370aab646R154